### ISSUE TITLE
Subroutines assigned via "On Enemy Sighted" are triggered by system and invisible objects

### ISSUE THREAD

**Issue by kABUSE1** [TheSuperHackers/GeneralsGamePatch#1819]
A subroutine assigned to a team via the option "On Enemy Sighted" will be triggered by literally any object that belongs to an enemy team.

This includes projectiles that get within range, system objects such as "emergency repair abilities" and pretty much everything else you can think of that belongs to the enemy player.

This is the reason why you can see a computer player move around some the vehicles in his base when you use for example usa scan ability on his CC.

**Comment by xezon** [comment]
Is this some kind of object?

**Comment by kABUSE1** [comment]
> Is this some kind of object?

What do you mean?

**Comment by xezon** [comment]
Does this feature reference anything in INI?

**Comment by kABUSE1** [comment]
I don't think you will find anything to influence this behaviour in ini level.

I imagine this logic to be super hardcoded somewhere deep down in the engine and I'd say this probably needs to be fixed via Thyme.


For individual units you can use a "Sighted" script where you can specify the exact type to look out for. So possibly a technical limitation that you can't do the same for teams, so that teams are just hardcoded to detect any enemy, regardless of object type or object status (e.g. stealth).

![grafik](https://user-images.githubusercontent.com/47530719/230771042-d71d4b2c-5008-4bb7-b680-ea4908b42697.png)
